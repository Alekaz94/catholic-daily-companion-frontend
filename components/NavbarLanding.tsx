import { View } from "react-native";
import LogoutButton from "./LogoutButton";
import NavButton from "./NavButton";
import { Ionicons } from '@expo/vector-icons';
import React from "react";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { AuthStackParamList } from "../navigation/types";
import { Layout } from "../styles/Layout";

type NavbarNavigationProp = NativeStackNavigationProp<AuthStackParamList>;

const NavbarLanding = () => {
    const navigation = useNavigation<NavbarNavigationProp>();

    return (
        <View style={Layout.navbarContainer}>
            <NavButton title={"Login"} screen={"Login"} />
            <NavButton title={"Sign up"} screen={"Signup"} />
        </View>
    );
}

export default NavbarLanding;