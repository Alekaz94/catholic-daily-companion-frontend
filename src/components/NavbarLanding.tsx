import { View, Image } from "react-native";
import React from "react";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { AuthStackParamList } from "../navigation/types";
import { Layout } from "../styles/Layout";
import cdc_transparent_black from "../assets/images/cdc_transparent_black.png"
import cdc_transparent from "../assets/images/cdc_transparent.png"
import { useAppTheme } from "../hooks/useAppTheme";
import { useTheme } from "../context/ThemeContext";

type NavbarNavigationProp = NativeStackNavigationProp<AuthStackParamList>;

const NavbarLanding = () => {
    const navigation = useNavigation<NavbarNavigationProp>();
    const theme = useAppTheme();
    const {isDark} = useTheme();

    return (
        <View style={[Layout.navbarContainer, {backgroundColor: theme.auth.navbar, flexDirection: "row", justifyContent: "center", alignContent: "center"}]}>
            <Image source={isDark ? cdc_transparent : cdc_transparent_black} style={{width: 250, height: 250, resizeMode: "contain"}} />
        </View>
    );
}

export default NavbarLanding;