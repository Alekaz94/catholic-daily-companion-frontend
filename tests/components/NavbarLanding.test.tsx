import { useNavigation, useRoute } from "@react-navigation/native"
import NavbarLanding from "../../components/NavbarLanding";
import { act, fireEvent, render } from "@testing-library/react-native";

type MockNavButtonProps = {
    title: string;
    screen: string;
}

jest.mock("@react-navigation/native", () => ({
    useNavigation: jest.fn(),
    useRoute: jest.fn(),
}));

describe("NavbarLanding", () => {
    const mockNavigate = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
        (useNavigation as jest.Mock).mockReturnValue({ navigate: mockNavigate });
    })

    it("renders correctly with Login and Sign up buttons", () => {
        const { getByText, getByTestId } = render(<NavbarLanding />);

        expect(getByText("Login")).toBeTruthy();
        expect(getByText("Sign up")).toBeTruthy();

        const loginButton = getByTestId("navbutton-Login");
        expect(loginButton.props.style).toEqual(
            expect.objectContaining({ backgroundColor: "#FAF3E0" }),
        );
    });

    it("includes both nav buttons with correct testIDs", () => {
        const { getByTestId } = render(<NavbarLanding />);
        expect(getByTestId("navbutton-Login")).toBeTruthy();
        expect(getByTestId("navbutton-Signup")).toBeTruthy();
    });

    it("navigates to Login and Signup when buttons are pressed", async () => {
        const { getByTestId } = render(<NavbarLanding />);
        
        await act(async () => {
            fireEvent.press(getByTestId("navbutton-Login"));
        });
        expect(mockNavigate).toHaveBeenCalledWith("Login");

        await act(async () => {
            fireEvent.press(getByTestId("navbutton-Signup"));
        });
        expect(mockNavigate).toHaveBeenCalledWith("Signup");
        expect(mockNavigate).toHaveBeenCalledTimes(2);
    });

    it("matches the snapshot", () => {
        const { toJSON } = render(<NavbarLanding />);
        expect(toJSON()).toMatchSnapshot();
    });

    it("renders buttons with correct text", () => {
        const { getByText } = render(<NavbarLanding />);
        expect(getByText("Login").props.children).toBe("Login");
        expect(getByText("Sign up").props.children).toBe("Sign up");
    })
})