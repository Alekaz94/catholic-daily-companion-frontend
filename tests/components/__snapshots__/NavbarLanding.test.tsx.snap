// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLanding matches the snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#D4AF37",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "elevation": 3,
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "space-around",
        "paddingHorizontal": 10,
        "paddingTop": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
      },
      {
        "backgroundColor": "#FAF3E0",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FAF3E0",
        "borderRadius": 8,
        "marginTop": 10,
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="navbutton-Login"
  >
    <Text
      style={
        [
          {
            "color": "#1A1A1A",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Login
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FAF3E0",
        "borderRadius": 8,
        "marginTop": 10,
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="navbutton-Signup"
  >
    <Text
      style={
        [
          {
            "color": "#1A1A1A",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Sign up
    </Text>
  </View>
</View>
`;
